redun/executors/k8s_utils.py

```diff
@@ -0,0 +1,254 @@
+        """
+        Loads k8s config if not loaded.
+        """
+        if not self._is_loaded:
+            try:
+                config.load_kube_config()
+            except ConfigException:
+                logger.warn(
+                    "config.load_kube_config() failed. "
+                    "Resorting to config.load_incluster_config()."
+                )
+                config.load_incluster_config()
+            self._is_loaded = True
+
+    def version(self) -> Tuple[int, int]:
+        """
+        Returns an API client support k8s version API.
+
+        https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/VersionApi.md
+        """
+        self.load_config()
+        version_info = client.VersionApi().get_code()
+        major = int(version_info.major.replace("+", ""))
+        minor = int(version_info.minor.replace("+", ""))
+        return major, minor
+
+    @property
+    def core(self) -> client.CoreV1Api:
+        """
+        Returns an API client support k8s core API.
+
+        https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md
+        """
+        if not self._core:
+            self.load_config()
+            self._core = client.CoreV1Api()
+        return self._core
+
+    @property
+    def batch(self) -> client.BatchV1Api:
+        """
+        Returns an API client supporting k8s batch API.
+
+        https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/BatchV1Api.md
+        """
+        if not self._batch:
+            self.load_config()
+            self._batch = client.BatchV1Api()
+        return self._batch
+
+
+def delete_k8s_secret(k8s_client: K8SClient, secret_name: str, namespace: str) -> None:
+    """
+    Deletes a k8s Secret. If Secret doesn't exist, it does not throw an error.
+    """
+    try:
+        k8s_client.core.delete_namespaced_secret(secret_name, namespace)
+    except ApiException as error:
+        if error.status != 404:
+            # If secret doesn't exist, then this is a no-op.
+            raise
+
+
+def create_k8s_secret(
+    k8s_client: K8SClient,
+    secret_name: str,
+    namespace: str,
+    secret_data: dict,
+    secret_type: str = "Opaque",
+) -> Any:
+    """
+    Creates a k8s Secret from a dict of key-value pairs.
+    """
+    # Build secret.
+    metadata = {"name": secret_name, "namespace": namespace}
+    data = {
+        key: b64encode(value.encode("utf8")).decode("utf8") for key, value in secret_data.items()
+    }
+    body = client.V1Secret("v1", data, False, "Secret", metadata, type=secret_type)
+
+    # Create secret.
+    try:
+        return k8s_client.core.create_namespaced_secret(namespace, body)
+    except ApiException as error:
+        if error.status == 409:
+            # Secret already exists, just patch it.
+            return k8s_client.core.replace_namespaced_secret(secret_name, namespace, body)
+        raise
+
+
+def create_resources(
```

Comment:
- [ ] Doesn't technically create the resource, only instantiates it locally. Do we need this function?